salt/docker/init.sls

```diff
@@ -40,21 +41,16 @@ docker:
     - append_if_not_found: True
     - require:
       - pkg: docker
-  # [inercia] when dockerd was already running and we require the
-  # service to be "service.running", Salt does not think it must
-  # restart it even when we say "watch these files", so we
-  # need this "cmd.run"...
-  cmd.run:
+  cmd.wait:
     - name: systemctl restart docker.service
-    - onlyif: systemctl status docker.service
+    - watch:
+      - file: /etc/systemd/system/docker.service.d/proxy.conf
```

Comment:
With the current code it will be restarted *always* on every orchestration/highstate.

Just check `journalctl -fu docker` and run a highstate on that machine, it is restarted everytime.